### PR TITLE
chore: enable dynamic font scaling

### PR DESCRIPTION
**Pull request by mapsandapps**
Enable dynamic font scaling & adjust a few things to look better at large font sizes.

See https://github.com/ionic-team/ionic-conference-app/pull/1805 for the Angular version.

You can use
```
    "@ionic/react": "7.4.2-dev.11695944163.13e7414e",
    "ionicons": "7.1.3-dev.11695669134.18369c36",
```
for testing

### CODE REVIEW

File: src/components/Menu.tsx [hunk 1]
```diff
@@ -132,7 +132,7 @@ const Menu: React.FC<MenuProps> = ({
             }}
           >
             <IonIcon slot="start" icon={hammer} />
-            Show Tutorial
+            <span className="tutorial-label">Show Tutorial</span>
```
Comment (liamdebeasi) [inline src/components/Menu.tsx:135]: The Angular app uses an `ion-label`. Should we use that here instead for consistency?

https://github.com/ionic-team/ionic-conference-app/blob/d74e556278264470856891c2f072186161aa5198/src/app/app.component.html#L102

File: src/components/Menu.css [hunk 2]
```diff
@@ -26,7 +26,8 @@ ion-menu.md ion-list-header {
   padding-right: 18px;
 
   text-transform: uppercase;
-  letter-spacing: .1em;
+  letter-spacing: 0.1em;
+  font-size: min(0.875rem, 32px);
```
Comment (liamdebeasi) [inline src/components/Menu.css:30]: The Angular conference app has a max font size of 36px: https://github.com/ionic-team/ionic-conference-app/pull/1805/files#diff-f17ac4f38c432797674ccd8854f9ce708922f6293f405ebc129414c13ec49c23R27

Any reason for the difference?

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

File: src/components/Menu.css (hunk 2 above)
Comment (mapsandapps) [reply src/components/Menu.css:30]: Yes, but I didn't look into what it was. 36 worked there, so I tried to use it here too, but it was too large. I can change that one to 32 for consistency, if you'd like.

Comment (liamdebeasi) [reply src/components/Menu.css:30]: Can you look into why there's a difference?

Comment (mapsandapps) [reply src/components/Menu.css:30]: Turns out they both need to be 32. I'm not sure what I changed on my end... or why 36 is working fine on your end 🤔

**Review by liamdebeasi — APPROVED** [review]

**Pull request merged**